src/docker/functions/container/create-container.ts

```diff
@@ -111,7 +111,7 @@ const getHealthCheck = (healthCheck?: HealthCheck): DockerodeHealthCheck | undef
   }
 
   return {
-    Test: ["CMD-SHELL", healthCheck.test],
+    Test: typeof healthCheck.test === "string" ? ["CMD-SHELL", healthCheck.test] : ["CMD", ...healthCheck.test],
```

Comment:
it's not expected for a user that the behaviour of the healthcheck should change if they provide their test as a string or an array of strings. I propose we change the type of the `test` field in `HealthCheck` to be as clear as possible and make this a breaking change. I'm thinking:

```typescript
export type HealthCheck = {
  test: ["CMD" | "CMD-SHELL", string];
  ...
}
```

as well as adding more information in the README on what CMD/CMD-SHELL means. What do you think?

Reply:
I would be disinclined to make it a breaking change, how about
```typescript
type HealthCheckCommon = {
  interval?: number;
  timeout?: number;
  retries?: number;
  startPeriod?: number;
};

type HealthCheckCommandShell = {
  test: string;
} & HealthCheckCommon;

type HealthCheckCommand = {
  testCmd: string[];
} & HealthCheckCommon;

export type HealthCheck = HealthCheckCommandShell | HealthCheckCommand;
```
and
```suggestion
    Test: 'testCmd' in healthCheck ? ["CMD", ...healthCheck.testCmd] : ["CMD-SHELL", healthCheck.test],
```
and to use it
```typescript
   const container = await customGenericContainer
      .withExposedPorts(8080)
      .withHealthCheck({
        testCmd: ["/usr/bin/wget", "--no-verbose", "--tries=1", "--spider", "http://localhost:8080/hello-world"],
      })
```
that way you can continue to use
```typescript
      customGenericContainer
        .withName(containerName)
        .withExposedPorts(8080)
        .withHealthCheck({ test: "exit 1" })
```
for the CMD-SHELL version.

Reply:
i would rather make a breaking change if it means having a simpler API and codebase. A user could provide both `test` and `testCmd` here and the behaviour is unexpected. They would also need to dig into the code to figure out what it means which isn't great. I think the code you've shared here compared to the interface change I proposed we could agree the latter is far simpler?

Reply:
ok, I'll make the breaking change